hail/src/main/scala/is/hail/expr/ir/GenericLines.scala

```diff
@@ -15,6 +18,14 @@ import scala.annotation.meta.param
 
 trait CloseableIterator[T] extends Iterator[T] with AutoCloseable
 
+object CloseableIterator {
+  def empty[T]: CloseableIterator[T] = new CloseableIterator[T] {
```

Comment:
a shame the iterator interface isn't properly covariant so this could just be a `val empty: CloseableIterator[AnyRef]`.